sys/src/9/ufs/ufs_harvey.c

```diff
@@ -48,3 +48,85 @@ releaseufsvnode(vnode *vn)
 	// TODO HARVEY - This assumes no sharing
 	free(vn);
 }
+
+/*
+ * Return the next vnode from the free list.
+ */
+int
+getnewvnode(const char *tag, MountPoint *mp, vnode **vpp)
+{
+	vnode *vp = nil;
+	//struct lock_object *lo;
+	//static int cyclecount;
+	//int error = 0;
+
+//alloc:
+	vp = (vnode *)smalloc(sizeof(vnode));
+
+	// TODO HARVEY Revisit locking of vnodes
```

Comment:
note: get the locking right from the start, your life is going to be pretty awful otherwise.